bloom-offchain/src/execution_engine/mod.rs

```diff
@@ -606,6 +572,38 @@ where
         }
     }
 
+    fn on_linkage_failure(&mut self, focus_pair: PR, orphans: NonEmpty<Either<CO, P>>)
+    where
+        SID: Eq + Hash + Copy + Display + Debug,
+        V: Eq + Hash + Copy + Display,
+        B: Clone + Debug,
+        MC: Clone,
+        PR: Eq + Hash + Copy + Display,
+        SO: SpecializedOrder<TOrderId = V>,
+        CO: Stable<StableId = SID> + Copy + Display,
+        P: Stable<StableId = SID> + Copy,
+        IX: StateIndex<EvolvingEntity<CO, P, V, B>>,
+        TLB: ExternalLBEvents<CO, P> + LBFeedback<CO, P> + Maker<MC>,
+    {
+        for fragment in &orphans {
+            let id = fragment.stable_id();
+            warn!("Linkage failed for {}", id);
+            self.index.eliminate(id);
```

Comment:
We are calling on_linkage_failure in case when we cannot find any `version` for stableId with prefixes from: LAST_PREDICTED_PREFIX, LAST_UNCONFIRMED_PREFIX, LAST_CONFIRMED_PREFIX, right?
Comes from 
```rust
match ExecutionRecipe::link(recipe, |id| {
    resolve_state(id, &self.index)
       .map(|Bundled(t, bearer)| (t.either(|b| b.version, |b| b.version), bearer))
})
```
But `eliminate` will delete version only in case of existence corresponding pair `prefix + stableId`

Reply:
Agree